### ISSUE TITLE
'The executable you are trying to debug requires elevation' endless loop

### ISSUE THREAD

**Issue by ioncodes** [x64dbg/x64dbg#1501]
- Debugger version: x64dbg v25 Mar 16 2017, 21:47:36
- Operating system version and Service Pack: Windows 10 x64

When I try to debug TeamSpeak, it will prompt me with said message. When I click on Yes, it will start x64dbg, ask me for administrative rights, when I accept me, it will just show me the message from before again. However, if I click again yes, it will restart, but not ask me for administrative rights, which means it does recognize the rights.
I tried via the right click context menu and starting x64dbg as admin and then load the file. No luck.

**Comment by mrexodia** [comment]
See [here](https://blogs.techsmith.com/inside-techsmith/devcorner-debug-uiaccess) for the reason. This is a limitation in Windows. You can work around this by using gflags (#816) to automatically attach x32dbg to the TeamSpeak client.

**Comment by mrexodia** [comment]
I added a workaround/detection mechanism to not come in an infinite loop. This will notify you of the `uiAccess="true"` status.

**Comment by ioncodes** [comment]
@mrexodia If I follow the guide (https://blogs.techsmith.com/inside-techsmith/devcorner-debug-uiaccess/) it still does not work.

**Comment by mrexodia** [comment]
No it will never work. You would have to change x64dbg to have uiAccess=true, re-sign it with a valid certificate and place it in C:\Program Files.

**Comment by Mattiwatti** [comment]
This should be quite easy to do, assuming Windows accepts self-signed certificates for uiAccess='true' applications. The way I do this is having two certificates: one self-signed root CA certificate placed in the trusted root store (this one is never actually used except to create the second certificate), and one certificate issued by the root CA (this one does not need to be installed anywhere). I've been signing executables this way for years now and it has always works fine (drivers are another story - compare `signtool verify /v /ca filename.exe` with `signtool verify /v /kp filename.sys`).
1. Sign x64dbg.exe or x32dbg.exe with the code signing certificate:
`signtool sign /v /ph /fd sha256 /s My /n "MyCertName" /sha1 "MyCertSha1Hash" /tr "http://sha256timestamp.ws.symantec.com/sha256/timestamp" /td sha256 x64dbg.exe`
2. 'Install' x64dbg in C:\Program Files by copying over the bin\x32 or bin\x64 directory to e.g. C:\Program Files\x64dbg.

**Comment by Mattiwatti** [comment]
@mrexodia: wouldn't the write access issue be solved by setting `requireAdministrator=true' as well? I mean, you're already giving a UAC prompt anyway.

**Comment by mrexodia** [comment]
Yes @Mattiwatti I meant that we cannot change it in x64dbg permanently because uiAccess=true and not being signed+admin location doesn't work nicely. `uiAccess=true` alone doesn't elevate...

**Comment by Mattiwatti** [comment]
Yes, that's true. You need quite a lot of pieces together to make the whole thing work...

It **could** be 'solved' by making a x64dbg_admin.exe (or _uiaccess.exe...) which is equivalent to the normal exe except having `<requestedExecutionLevel level='requireAdministrator' uiAccess='true' />`. Since x64dbg is normally digitally signed (unless you are building from source), that would solve some of the issues. You'd still need to put x64dbg in Program Files however. But I don't really think the added confusion of having two separate .exe files would help very much, especially since 99% of users won't encounter this scenario. (But it might wrongly make people think they need the second .exe to debug programs as administrator, which is false.)

I found this random snippet (have not tested) on [https://social.msdn.microsoft.com/Forums/windowsdesktop/en-US/4d2e1358-af95-4f4f-b239-68ec7e2525a9/uiaccess-in-manifest-files?forum=windowssecurity](https://social.msdn.microsoft.com/Forums/windowsdesktop/en-US/4d2e1358-af95-4f4f-b239-68ec7e2525a9/uiaccess-in-manifest-files?forum=windowssecurity):
"_Regardless of the state of the uiAccess flag, your application will always be able to send messages/drive input to windows of applications running at privilege levels equal to or less than your own privilege level._"
So potentially it would be possible to create a non-admin user, run TeamSpeak as that with right click -> run as, and then debug it. But then we're back to the original response, which is to just launch the exe first and then attach, which is by far the easiest solution anyway.

**Comment by mrexodia** [comment]
Yeah I was thinking about a separate executable, but another downside of that is that the settings won't be kept so that will lead to even more confusion. 😄 

@ioncodes if you want to debug the executable on launch #816 will handle that case...

**Comment by Mattiwatti** [comment]
Btw, I just remembered it **is** possible to have an external manifest, i.e. x64dbg.exe.manifest. (removing the current one from the PE resource section). I'm not sure if this approach would work however, since it seems rather weak from a security POV to have the application's manifest editable by anyone...

OTOH, I do remember a case where I had an application that depended on a 2 MB Armadillo DLL for licensing stuff, but they didn't actually **need** it (in the API/programming sense) and none of the binaries were linked against it, so instead they added it as a 'dependentAssembly' in an external .manifest file. I circumvented this complicated DRM scheme by deleting the .manifest file, which worked...
